### DISCUSSION TITLE
Update your extensions from Manifest V2 to V3

### DISCUSSION THREAD

**Discussion by nagachaitanyalokam** [microsoft/MicrosoftEdge-Extensions#27] [Announcements]
Microsoft announced the [decision to embrace Manifest V3](https://blogs.windows.com/msedgedev/2020/10/14/extension-manifest-chromium-edge/) to help reduce web fragmentation for all developers and enhance privacy, security, and performance for end-users.

### **From 11 July 2022:**

- Microsoft Partner Center will **no longer accept new Manifest V2 extensions** with visibility set as `Hidden` or `Public`. Developers must use Manifest V3 to publish new extensions to the Microsoft Edge Add-ons website.
- You may continue to update existing Manifest V2 extensions. For more details on the deprecation of Manifest V2 extensions, visit [Overview and timelines for migrating to Manifest V3](https://docs.microsoft.com/microsoft-edge/extensions-chromium/developer-guide/manifest-v3).

Migrating from Manifest V2 to V3 will bring several structural changes to how the browser handles extensions. Manifest V3 is an initiative of the Chromium project.

Here is the link to the [official announcement](https://aka.ms/MicrosoftEdge_ManifestV2_Sunset_Announcement) made on Microsoft Edge Insider Tech community

Check out the below resources to learn more about the upcoming changes:

**Documentation:**
1. [Overview and timelines for migrating to Manifest V3](https://docs.microsoft.com/microsoft-edge/extensions-chromium/developer-guide/manifest-v3)
2. [Manifest file format for extensions](https://docs.microsoft.com/microsoft-edge/extensions-chromium/getting-started/manifest-format)

**Roadmap:**
[Microsoft Edge Add-ons Roadmap](https://docs.microsoft.com/en-us/microsoft-edge/extensions-chromium/whats-new/roadmap)

Let us know your feedback / challenges in the comments.

**Comment by ystolzenburg** [comment]
How do you respond to concerns from the main uBlock Origin developer (https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-456179825)? This extension has over 10 mio users on MS Edge and won't work after V3 becomes mandatory.

**Reply by DarkShadowMD** [reply]
I am also concerned about this. Very little has been told. If Edge doesn't support full functionality, in fact, privacy is compromised...

**Reply by xlollomanx** [reply]
I'm too curious about how this will end.
Seeing their roadmap they will definetly drop v2 one day:  

> When this change is released, Microsoft Partner Center will not accept updates to existing Manifest V2 Microsoft Edge Add-ons. Microsoft Edge will no longer support or run Manifest V2 Microsoft Edge Add-ons.
> 
> Note that by using Microsoft Edge Enterprise Policies, Enterprise Accounts can allow Manifest V2 Microsoft Edge Add-ons to run and operate on Microsoft Edge.

I think microsoft have done a good job with edge till now, but if ublock will not work anymore I'll never use edge again and I don't think I'm the only one.

**Reply by ashishnirvana1** [reply]
We are closely watching this as well. uBlock has published an experimental MV3 extension (title uBlock Origin Lite) to Chrome. You may want to try that out and provide feedback if you see any specific deficiencies.

**Reply by SHAGGAR** [reply]
It seems like ublock origin lite isnt gonna work out.  It doesnt work anywhere near as well and to get it to approach the level of v2, the browser spams the user with nagging permissions prompts that will train the user to always click yes. 

We need to stop having discussions about "try it out and provide feedback" because everyone knows what the outcome here is. 
 V3 is not about security or privacy, its about Google wanting to break adblockers because Google is an advertising company.  

Edge has created a bunch of goodwill for Microsoft, and now you guys are looking to throw it away by mandating V3 and bringing ads back into our lives.  Ads are the largest security and privacy threat that users will face on the web and content blockers like ublock protect users more than any of these V3 changes will.  

Microsoft could win alot of hearts and minds by retaining v2 support forever.  Or by creating a feature superset for v3 (lets call it v3+) that allows extensions to bring in v2 content blocking features.  Extensions could check for feature availability in order to retain browser compatibility.

In addition if you're actually worried about extensions causing privacy or security problems (and lets be honest you arent) you could have an extensions approval process whereby extensions are validated by microsoft against threats and then signed before being deployed to the store.  These signed and approved extensions could be granted higher permissions.  Effectively treating extensions like apps in the microsoft store or ios app store.  Unsigned extensions would still be allowed, but they'd be subject to the stricter controls of v3.

IDK, think about it.  The current plan stinks.

**Reply by Jackenmen** [reply]
> uBlock has published an experimental MV3 extension (title uBlock Origin Lite) to Chrome. You may want to try that out and provide feedback if you see any specific deficiencies.

The lack of the element picker is a bit annoying since sometimes I want to get rid of some small thing showing up on a webpage that's annoying me, this doesn't generally mean an advertisement, I mostly use it to remove overlays and that's not possible with uBlock Origin Lite. Other than that, only being able to update filters on extension updates means that you end up seeing ads more often because the approval process of extension updates tends to be slower ([especially for extensions with broad host permissions](https://developer.chrome.com/docs/webstore/review-process/#review-time-factors)). I still have to use it longer to figure out all the quirks, will try to report back when I have any more feedback.

**Comment by gdh1995** [comment]
Will MS Edge also kill background service workers every 5 minutes like what Chrome does? Any way to configure the time range or disable it?

**Updated**: I've found https://stackoverflow.com/a/66618269

**Comment by lsmith77** [comment]
If DOM access is removed, then even Microsoft's own "Microsoft Editor" extension will no longer work.

**Comment by marcussacana** [comment]
The day that the Manifest v3 became required, will be the same one that I change my browser.
there's nothing that can change my mind.

**Reply by QinCai-rui** [reply]
I already switched to Firefox. Almost everything works fine! :)))

**Comment by joeyboli** [comment]
Disable support for v2 and im moving to firefox or brave.

**Reply by marcussacana** [reply]
Same, I'm in firefox already, I recommend it feels very light, also a good opportunity to quit chromium.
![image](https://github.com/user-attachments/assets/08bfe307-36cd-44ce-a7a3-75e65629b05d)

**Reply by QinCai-rui** [reply]
Same! luckily uBO is working 100%!!

**Comment by marypcbuk** [comment]
is there a timeline for updating the timeline on the MV2 MV3 shift?

**Reply by QinCai-rui** [reply]
I really hope Microsoft is thinking about it and (hopefully!) continue to support MV2 extensions. Fingers crossed...

**Comment by mesutcelik** [comment]
What is the timeline on the MV2 to MV3 switch? Is there any update on the roadmap?

**Reply by marypcbuk** [reply]
I know there was some confusion when people who had got their V2 extensions from the Chrome store rather than the Edge store couldn't turn them back on when they were turned off in the recent update; this would be a great time for some clarity, even if it's 'still a long way off and TBD'

**Reply by ManikanthMSFT** [reply]
Hi @mesutcelik, @marypcbuk, thank you for reaching out with your question regarding the timeline for the MV2 to MV3 switch and any updates on the roadmap.

Currently, there is no time frame to convert from MV2 to MV3. If we get a get the road map we will update here.

We appreciate your understanding and cooperation.

**Reply by SHAGGAR** [reply]
> Hi @mesutcelik, @marypcbuk, thank you for reaching out with your question regarding the timeline for the MV2 to MV3 switch and any updates on the roadmap.
> 
> Currently, there is no time frame to convert from MV2 to MV3. If we get a get the road map we will update here.
> 
> We appreciate your understanding and cooperation.

Sounds like its not too late to change your mind on supporting Google's banning of effective adblockers.  You could keep mv2 around forever, or maybe limit its usage to ublock origin and other well known adblockers, or maybe its time to consider a fully integrated browser adblocker.  Something that supports the same content interception/replacement and custom user lists/element picking that mv2 adblockers provide, but native to the browser.

**Comment by alberquin** [comment]
Please DON'T implement Manifest V3.

**Comment by demirten** [comment]
Continuing support for Manifest V2 extensions would be a valuable step toward accommodating enterprise customers who still depend on policy-installed MV2 extensions.

Alternatively, addressing the current limitations of the webRequestBlocking feature in Manifest V3—specifically for policy-installed extensions—could offer a practical path forward.

As stated in the documentation:

> "webRequestBlocking": Required to register blocking event handlers. As of Manifest V3, this is only available to policy-installed extensions.

This indicates that enterprise environments can still utilize webRequestBlocking under MV3 through policy-based deployment. However, the existing implementation lacks support for synchronous operations within request handlers. Since synchronous XMLHttpRequest has been deprecated and fetch is inherently asynchronous, it’s currently not feasible to perform real-time lookups or native messaging within the blocking callback.

If webRequest callbacks were enhanced to support fully asynchronous execution (e.g., via async/await or Promises), Manifest V3 would become significantly more viable for enterprise use cases. This would allow policy-installed extensions to continue performing real-time, dynamic request filtering without relying on MV2 capabilities.

**Comment by SHAGGAR** [comment]
Google's ad business was declared an illegal monopoly by the DOJ which also is requesting Google sell off chrome.  Chromium is the most used browser project on the internet and Google's dominant control over the chromium project allows them to force through features that promote their own non-browser businesses: in the case with MV3, the business being supported is an advertising monopoly. 

MV3 only exists to break adblockers.  It was created to promote Google's illegal ad monopoly at the expense of users.  

I don't see how anyone could in good faith support Google committing and promoting MV3 and other pro-advertiser features in Chromium in furtherance of their illegal monopolies.  Microsoft Edge and other Chromium downstream projects should reject MV3 and/or overhaul it to remove Google's anti-consumer/pro-ad monopoly features.

**Comment by spacecheeserocks** [comment]
I'd like to request some clarification from the Edge team regarding MV2 deprecation.

The [current Manifest V3 timeline](https://learn.microsoft.com/en-us/microsoft-edge/extensions/developer-guide/manifest-v3#manifest-timeline-for-microsoft-edge-and-partner-center) (as of 4 August 2025) still marks the MV2 deprecation as TBD, providing very little assurance or clarity about how long MV2 extensions will still be supported.

Google Chrome v139 is currently released to 0.5% of Chrome Stable Users, and is [scheduled for regular release on 5 August 2025](https://chromiumdash.appspot.com/schedule). I can confirm that Chrome v139 definitely has completely removed the ability to install MV2 extensions (even with policy, MV2 extensions are simply ignored with no warning). Chrome v138 was the last version to support MV2 extensions.

I can also confirm experimentally that despite the above, Edge 139 still supports MV2 extensions (verified by using Edge Beta, currently v139). 

However, with no further announcement from Microsoft, this raises concerns that MV2 extensions will "suddenly" be disabled in an upcoming release. My assumption is that Google / The Chromium Team will have very little incentive to maintain or preserve MV2 related code, and that it will be stripped from the Chromium source entirely in an upcoming update - and that it will not be trivial for the Edge team to maintain this compatibility by themselves.

I am mostly asking this question from an IT Administration perspective, as we currently rely on an MV2 extension in our company. We currently don't know if Microsoft will continue to support MV2 for the forseeable future, or if we're going to be suddenly given a very short notice to respond (or switch to Firefox!).

**Comment by allentown521** [comment]
Is there an up-to-date timeline for when the MV2 extension will no longer be able to run in Edge?

**Comment by hlcigarashimotoki** [comment]
Is the timing for the deprecation of MV2 extensions in Edge still undecided?
It doesn't seem to be mentioned in the currently released versions.
We need to make business plans, so could you at least tell us the planned timeframe?
For example:
January-March 2026, 
April-June 2026, etc.

**Reply by marypcbuk** [reply]
Perhaps now that Google has changed so many of its plans to replace third party cookies its time for Microsoft to reconsider and put M3 on official indefinite hold (rather than the unofficial indefinite hold it's at currently)…
